test/utils/padder/padder.go

```diff
@@ -0,0 +1,212 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package padder
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	nrtv1alpha1 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha1"
+	"github.com/openshift-kni/numaresources-operator/test/utils/fixture"
+	nrtutil "github.com/openshift-kni/numaresources-operator/test/utils/noderesourcetopologies"
+	"github.com/openshift-kni/numaresources-operator/test/utils/objects"
+	"github.com/openshift-kni/numaresources-operator/test/utils/objects/wait"
+)
+
+// This package allows to control the amount of available allocationTarget under the nodes.
+// it's doing that by "padding" the nodes with guaranteed pods until node reaches the desired amount of available allocationTarget.
+//
+// Usage example:
+// pad := New(client, "test-ns")
+// pad.Nodes(2).UntilAvailableIsResource(corev1.ResourceCPU, "3").UntilAvailableIsResource(corev1.ResourceMemory, "100Mi").Pad()
+// or
+// rl := corev1.ResourceList {
+//		corev1.ResourceCPU: "10",
+//		corev1.ResourceMemory: "100Mi"
+//	}
+// pad.Nodes(2).UntilAvailableIsResourceList(rl).Pad()
+
+const PadderLabel = "nrop-test-pad-pod"
+
+type Padder struct {
+	// Client defines the API client to run CRUD operations, that will be used for testing
+	Client    client.Client
+	namespace string
+	*padRequest
+}
+
+type padRequest struct {
+	nNodes           int
+	allocationTarget corev1.ResourceList
+}
+
+// New return new padder object
+// ns is where padding pods will get deployed
+func New(cli client.Client, ns string) (*Padder, error) {
+	return &Padder{
+		Client:     cli,
+		namespace:  ns,
+		padRequest: &padRequest{allocationTarget: make(corev1.ResourceList)},
+	}, nil
+}
+
+// Nodes number of nodes to pad
+func (p *Padder) Nodes(n int) *Padder {
+	p.padRequest.nNodes = n
+	return p
+}
+
+// UntilAvailableIsResource will pad pods into nodes until reach the desired available allocationTarget
+func (p *Padder) UntilAvailableIsResource(resName corev1.ResourceName, quantity string) *Padder {
+	quan := resource.MustParse(quantity)
+	p.allocationTarget[resName] = quan
+	return p
+}
+
+// UntilAvailableIsResourceList is like UntilAvailableIsResource but with a complete ResourceList as a parameter
+func (p *Padder) UntilAvailableIsResourceList(resources corev1.ResourceList) *Padder {
```

Comment:
do we have a clear usecase for the two `UntilAvailableIs*` APIs or are we adding them "just in case"?

Reply:
We need at least one of them :) 
We have both of them for convenience

Reply:
ack